algorithms/src/polycommit/data_structures/powers.rs

```diff
@@ -0,0 +1,375 @@
+    #[cfg(feature = "wasm")]
+    pub fn download_up_to(&self, degree: usize) -> Result<()> {
+        let degrees_to_download = self.get_degrees_to_download(degree);
+        for d in &degrees_to_download {
+            if let Some(link) = URLS.get(d) {
+                let buffer = alloc::sync::Arc::new(parking_lot::RwLock::new(vec![]));
+                let url = String::from(link);
+
+                // NOTE(julesdesmit): I'm leaking memory here so that I can get a
+                // static reference to the url, which is needed to pass it into
+                // the local thread which downloads the file.
+                let buffer_clone = alloc::sync::Arc::downgrade(&buffer);
+                // NOTE(julesdesmit): We spawn a local thread here in order to be
+                // able to accommodate the async syntax from reqwest.
+                wasm_bindgen_futures::spawn_local(async move {
+                    let content = reqwest::get(url).await.unwrap().text().await.unwrap();
+
+                    let buffer = buffer_clone.upgrade().unwrap();
+                    buffer.write().extend_from_slice(content.as_bytes());
+                    drop(buffer);
+                });
+                // Recover the bytes.
+                let buffer = alloc::sync::Arc::try_unwrap(buffer).unwrap();
+                let buffer = buffer.write().clone();
+                let mut file = self.file.write();
+                file.seek(SeekFrom::End(0))?;
+                file.write_all(&buffer)?;
+            } else {
+                return Err(anyhow!("incorrect degree selected - {}", degree));
+            }
+        }
+
+        *self.degree.write() = *degrees_to_download.last().unwrap();
+        self.regenerate_powers_of_beta_times_gamma_g();
+
+        Ok(())
+    }
+
+    fn regenerate_powers_of_beta_times_gamma_g(&self) {
+        let mut alpha_powers_g1 = vec![];
+        let mut reader = BufReader::new(*POWERS_TIMES_GAMMA_G);
+        for _ in 0..NUM_POWERS_TIMES_GAMMA_G {
+            let power: E::G1Affine =
+                FromBytes::read_le(&mut reader).expect("hardcoded powers times gamma g should be well-formed");
+            alpha_powers_g1.push(power);
+        }
+
+        let mut alpha_tau_powers_g1 = BTreeMap::new();
+        for (i, power) in alpha_powers_g1.iter().enumerate().take(3) {
+            alpha_tau_powers_g1.insert(i, *power);
+        }
+        alpha_powers_g1[3..].iter().chunks(3).into_iter().enumerate().for_each(|(i, c)| {
+            // Avoid underflows and just stop populating the map if we're going to.
+            if self.len() - 1 > (1 << i) {
+                let c = c.into_iter().collect::<Vec<_>>();
+                alpha_tau_powers_g1.insert(self.len() - 1 - (1 << i) + 2, *c[0]);
+                alpha_tau_powers_g1.insert(self.len() - 1 - (1 << i) + 3, *c[1]);
+                alpha_tau_powers_g1.insert(self.len() - 1 - (1 << i) + 4, *c[2]);
+            }
+        });
```

Comment:
What is `alpha` and what is `tau` here?

Reply:
Not entire sure - I copied the code from some of the setup scripts. I can change the naming if needed, but I figured I'd stick with it for now!